File: baekjoon/python/9375.py
import sys; input = sys.stdin.readline
import collections

for _ in range(int(input())):
    cloth_dic = collections.defaultdict(int)

    count = int(input())
    for _ in range(count):
        item, key = input().split()
        cloth_dic[key] += 1
    
    result = 1
    for c in cloth_dic.values(): result *= c + 1
    result -= 1
    print(result)

File: baekjoon/javascript/9375.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", line => {
  input.push(line);
}).on("close", () => {
  let testCount = input.shift();
  while (testCount--) {
    let count = input.shift();
    const cloth = {};
    for (let i = 0; i < count; i++) {
      const [_, key] = input.shift().split(" ");
      if (cloth[key]) cloth[key]++;
      else cloth[key] = 1;
    }
    let result = 1;
    for (const i of Object.values(cloth)) result *= i + 1;
    console.log(result - 1);
  }
});
